src/TrackGenerator.cpp

```diff
@@ -398,8 +376,13 @@ FP_PRECISION TrackGenerator::getFSRVolume(int fsr_id) {
 
 
 /**
- * @brief Finds and returns the maximum optical length amongst all segments.
- * @return the maximum optical path length
+ * @breif Calculates and returns the maximum optcial length for any segment
+ *        in the Geomtry.
+ * @details The _max_optical_length value is recomputed, updated, and returned.
```

Comment:
Again, I lean against referencing a private variable `_max_optical_length` in a public-facing comment, and would instead opt for simply saying plain English "max optical path length"